src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs

```diff
@@ -0,0 +1,282 @@
+            using (var projectClient = connection.GetClient<ProjectHttpClient>())
+            using (var policyClient = connection.GetClient<PolicyHttpClient>())
+            using (var gitRepositoryClient = connection.GetClient<GitHttpClient>())
+            {
+                var projects = projectClient.GetProjects();
+
+                foreach (var project in projects.Result)
+                {
+                    if (!policyDefinition.IsProjectAllowed(project))
+                    {
+                        Log.Debug($"Skipping project \"{project.Name}\", because it's not listed in the allowed projects list");
+                        continue;
+                    }
+
+                    Log.Info($"Starting project: {project.Name}");
+
+                    var types = policyClient.GetPolicyTypesAsync(project.Id).Result;
+
+                    var policyConfigurations = policyClient.GetPolicyConfigurationsAsync(project.Id);
+                    var grouppedRepositories =
+                        policyConfigurations.Result
+                            .Where(x => !policyDefinition.IgnoreType(x.Type.Id, types))
+                            .GroupBy(x => x.GetRepositoryId());
+
+                    var repositories = gitRepositoryClient.GetRepositoriesAsync(project.Id);
+
+                    foreach (var repository in repositories.Result)
+                    {
+                        if (!policyDefinition.IsRepositoryAllowed(repository))
+                        {
+                            Log.Debug($"Skipping repository \"{repository.Name}\", because it's not listed in the allowed repositories list");
+                            continue;
+                        }
+
+                        var relevantPolicies = this.GetPoliciesForRepository(policyDefinition.Policies, project, repository);
+                        var tfsPolicies = grouppedRepositories.FirstOrDefault(x => x.Key == repository.Id.ToString());
+
+                        this.HandleChanges(policyDefinition.AllowDeletion, policyClient, project.Id, repository, relevantPolicies, tfsPolicies, types);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Creates a policy in TFS.
+        /// </summary>
+        /// <param name="policyClient">Policy client</param>
+        /// <param name="types">Types</param>
+        /// <param name="projectId">Team project id</param>
+        /// <param name="repository">Git repository</param>
+        /// <param name="currentPolicy">Branch policy</param>
+        /// <param name="policy">Policy</param>
+        protected abstract void CreatePolicy(
+            PolicyHttpClient policyClient,
+            IEnumerable<PolicyType> types,
+            Guid projectId,
+            GitRepository repository,
+            BranchPolicies currentPolicy,
+            Policy policy);
+
+        /// <summary>
+        /// Updates a policy in TFS.
```

Comment:
Use AzureDevOps instead of TFS, as on-prem product is now also called Azure DevOps Server